factoryWithTypeCheckers.js

```diff
@@ -364,6 +365,36 @@ module.exports = function(isValidElement, throwOnDirectAccess) {
     return createChainableTypeChecker(validate);
   }
 
+
+  function createElementWithTypeChecker(expectedType) {
+    var ACCEPTABLE_TYPES_OF_EXPECTED_TYPES = ['string', 'function'];
+
+    if (ACCEPTABLE_TYPES_OF_EXPECTED_TYPES.indexOf(typeof expectedType) === -1) {
```

Comment:
instead of making an array and using indexOf, this should repeat `typeof` here - it's actually optimized to be faster anyways. However, instead, why not:
```suggestion
    if (!isValidElementType(expectedType)) {
```
? (using `isValidElement` from `react-is`)?